pkgs/by-name/mq/MQTT-Explorer/package.nix

```diff
@@ -0,0 +1,181 @@
+    popd
+
+    pushd backend
+    fixup-yarn-lock yarn.lock
+    yarn config --offline set yarn-offline-mirror $offlineCacheApp
+    yarn install --offline --frozen-lockfile --ignore-engines --ignore-scripts --no-progress
+    popd
+
+    patchShebangs {node_modules,app/node_modules,backend/node_modules}
+
+    cp -r ${electronDist} electron-dist
+    chmod -R u+w electron-dist
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    tsc && cd app && yarn --offline run build && cd ..
+
+    yarn --offline run electron-builder --dir \
+      -c.electronDist=electron-dist \
+      -c.electronVersion=${electron.version}
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ${lib.optionalString (!stdenv.isDarwin) ''
+      mkdir -p "$out/share/MQTT-Explorer"/{app,icons/hicolor}
+
+      cp -r build/*-unpacked/{locales,resources{,.pak}} "$out/share/MQTT-Explorer/app"
+
+      for file in res/appx/Square44x44Logo.targetsize-*_altform-unplated.png; do
+
+        size=$(echo "$file" | sed -n 's/.*targetsize-\([0-9]*\)_altform-unplated\.png/\1/p')
+
+        install -Dm644 \
+          "$file" \
+          "$out/share/icons/hicolor/''${size}x''${size}/apps/MQTT-Explorer.png"
+      done
+
+      makeWrapper '${electron}/bin/electron' "$out/bin/MQTT-Explorer" \
+        --add-flags "$out/share/MQTT-Explorer/app/resources/app.asar" \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+        --set-default ELECTRON_FORCE_IS_PACKAGED 1 \
+        --set-default ELECTRON_IS_DEV 0 \
+        --inherit-argv0
+    ''}
+
+    ${lib.optionalString stdenv.isDarwin ''
+      mkdir -p $out/{Applications,bin}
+      mv "build/mac/MQTT Explorer.app" $out/Applications
+
+      makeWrapper "$out/Applications/MQTT Explorer.app/Contents/MacOS/MQTT Explorer" \
+        $out/bin/MQTT-Explorer
+    ''}
+
+    runHook postInstall
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+    export ELECTRON_OVERRIDE_DIST_PATH=electron-dist/
+
+    yarn test:app --offline
+    yarn test:backend --offline
+
+    unset ELECTRON_OVERRIDE_DIST_PATH
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = meta.mainProgram;
+      icon = "MQTT-Explorer";
+      desktopName = "MQTT Explorer";
+      genericName = "MQTT Protocol Client";
+      comment = meta.description;
+      type = "Application";
+      categories = [
+        "Development"
+        "Utility"
+        "Network"
+      ];
+      startupWMClass = "MQTT-Explorer";
+    })
+  ];
+
+  meta = with lib; {
+    description = "An all-round MQTT client that provides a structured topic overview";
+    homepage = "https://github.com/thomasnordquist/MQTT-Explorer";
+    changelog = "https://github.com/thomasnordquist/MQTT-Explorer/releases/tag/v${version}";
+    license = licenses.cc-by-nc-nd-40;
```

Comment:
Upstream states "CC-BY-ND-4.0" (https://github.com/thomasnordquist/MQTT-Explorer/blob/03ba43038cb43d267371f36d46d0ce8769dd1072/package.json#L77C15-L77C27), but there is no `cc-by-nd-40` one in licenses, perhaps it should be added.